US-Trustee-Program/Bankruptcy-Oversight-Support-Systems
# CAMS-14 sql with managed identity

# Purpose

Switch from using SQL auth to User Managed Identity to grant functionapp access to sql server database
# Major Changes

Describe what has changed.

- Update and refactoring existing database client and config logic to support connecting with managed identity. Also, improved on error logging.
- Update GHA deployment and biciep to provision managed identity for read access and assign to functionapp instance as a user identity
- Check for sql database access via healthcheck endpoint
- Additional unit tests added
- Documentation updated

# Testing/Validation

- Confirm database access via user managed identity from functionapp instance
- Verify automation of managed identity creation and assignment to functionapp
- Validate configuration in Flexion environment only

# Notes

- Manual step needed to grant access to managed identity. Because this is SQL server will need administrator with privileges to execute queries. Steps documented.
